src/server.cpp

```diff
@@ -167,7 +171,12 @@ void CHighPrecisionTimer::Start()
 #endif
 
         // start thread
+#ifndef SCHED_RR
         QThread::start ( QThread::TimeCriticalPriority );
+#else
+        // priorty is "ignored" on linux: actually broken
```

Comment:
But you set "QThread::NormalPriority" in your jam recorder. Why is that working and the "QThread::TimeCriticalPriority" not? That may be an issue with user rights? Maybe the Linux user running the Jamulus server does not have enough rights to get this priority?
But maybe this is a bug in Qt. Have you tried to use a newer Qt version?

Reply:
It's not "working" as such -- passing _any_ value drops real time and assumes SCHED_OTHER with nice zero (i.e. default scheduling values).

There's a statement in the QThread documentation that it doesn't work on Linux.
https://doc.qt.io/qt-5/qthread.html#start
>  In particular, the priority will be ignored on systems that do not support thread priorities (such as on Linux, see the sched_setscheduler documentation for more details).
Now, I'd say it is a _bug_ in Qt because it's causing an undesirable change rather than being "ignored".

I'm investigating workarounds.

https://man7.org/linux/man-pages/man2/setpriority.2.html works on threads, as far as I understand it -- pass the thread ID and it'll set the nice value (for SCHED_OTHER and SCHED_BATCH).
https://man7.org/linux/man-pages/man2/sched_setparam.2.html works on threads -- but allows SCHED_FIFO and SCHED_RR priorities to be set

I've not worked with these calls before, so I'm struggling to get a _working_ example -- and I want it to _not_ require "root", so it should work with CAP_SYS_NICE:
https://man7.org/linux/man-pages/man7/capabilities.7.html

Reply:
> passing any value drops real time and assumes SCHED_OTHER with nice zero

That is very interesting... According to the Qt description it should ignore the value if it is not supported on the OS. So that seems really be a Qt issue. What Qt version do you currently use?

Reply:
Qt 5.9 here.

Reply:
Looks like someone is maintaining backports for Ubuntu 18.04 LTS:
https://launchpad.net/~beineri
with 5.14.2 being the latest:
https://launchpad.net/~beineri/+archive/ubuntu/opt-qt-5.14.2-bionic

(Later ones for later versions of Ubuntu look like they were only released recently, so I'm hopeful for 18.04 LTS.  My 18.04 "end of life" plan kicks in later this year.)

Reply:
Qt 5.9 is quite old. Just yesterday I upgraded my Ubuntu laptop to the newest Ubuntu version without any problems. I can recommend to do it.

Reply:
I'll be taking it but not until I've a a good couple of free days - which means not for a couple of months, which fits nicely with my preference for leaving it a while after the LTS .1 has been out.